adobe/react-spectrum
# fix(#3306): useComboBox: Escape key should continue propagation, when appropriate, to close parent Overlay

Closes #3306 

## ✅ Pull Request Checklist:

- [x] Included link to corresponding React Spectrum GitHub Issue #3306.

## 📝 Test Instructions:

1. open https://reactspectrum.blob.core.windows.net/reactspectrum/821a9e31789bb3021d33ba0e97fcb532669f75a9/storybook/index.html?path=/story/combobox--within-a-dialog&providerSwitcher-toastPosition=bottom
2. Tab to set focus to the DialogTrigger Button
3. Press Space or Enter to open the Dialog
4. Tab to set focus to the ComboBox within the Dialog
5. Enter some arbitrary text into the ComboBox input
6. Press Esc key to clear the input value.
7. Press Esc key again with the input value empty.
8. Dialog should close, restoring focus to the DialogTrigger Button
9. Press Space or Enter to open the Dialog
10. Tab to set focus to the ComboBox within the Dialog
11. Press ArrowDown to expand the ComboBox dropdown listbox.
12. Use Arrow keys to navigate items in the listbox
13. Select an item by pressing Enter.
14. Listbox dropdown should close, restoring focus to the input containing the selected value.
15. Press Esc key, which with a selected key should close the Dialog, restoring focus to the DialogTrigger Button.


16. https://reactspectrum.blob.core.windows.net/reactspectrum/821a9e31789bb3021d33ba0e97fcb532669f75a9/storybook/index.html?path=/story/combobox--within-a-dialog-allowscustomvalue-true&providerSwitcher-toastPosition=bottom
17. Tab to set focus to the DialogTrigger Button
18. Press Space or Enter to open the Dialog
19. Tab to set focus to the ComboBox within the Dialog
20. Enter some arbitrary text into the ComboBox input
21. Press Esc key.
22. The Dialog should close, because with `allowsCustomValue`, any value is acceptable within the ComboBox, so Esc key event should `continuePropagation`.

## 🧢 Your Project:

Adobe/Accessibility
